core/src/main/java/com/graphhopper/routing/weighting/FlexModelWeighting.java

```diff
@@ -84,13 +84,14 @@ public double getMinWeight(double distance) {
 
     @Override
     public double calcWeight(EdgeIteratorState edge, boolean reverse, int prevOrNextEdgeId) {
-        double seconds = calcSeconds(edge, reverse, prevOrNextEdgeId);
+        double distance = edge.getDistance();
+        double seconds = calcSeconds(distance, edge, reverse, prevOrNextEdgeId);
         if (Double.isInfinite(seconds))
             return Double.POSITIVE_INFINITY;
-        return (seconds + edge.getDistance() * distanceFactor) / priorityConfig.calcPriority(edge, reverse, prevOrNextEdgeId);
+        return (seconds + distance * distanceFactor) / priorityConfig.calcPriority(edge, reverse, prevOrNextEdgeId);
```

Comment:
Why does FlexModelWeighting not inherit from AbstractWeighting ? If it did maybe the distance could be handled this way somehow ? The other reason I am asking is that with #1820 the turn costs will be handled in AbstractWeighting (unless we find a better way to do this not entirely sure) so if we do not want to duplicate the turn cost handling it would help to have FlexModelWeighting extends AbstractWeighting

Reply:
Or maybe to put it another way: Maybe there should be some parent class that handles turn costs (and distance ?). AbstractWeighting is probably not a good choice atm because of access/avgSpeedEnc (?).

Reply:
It should be no problem to extend from it. Initially I wanted to avoid to rely on any previous logic like the FlagEncoder requirement (obviously less successful ;)). But it helped me to better understand what to do.

> AbstractWeighting is probably not a good choice atm because of access/avgSpeedEnc (?).

What do you mean here?

Reply:
> What do you mean here?

I thought maybe you did not extend from AbstractWeighting because of the access/speedEnc fields

Reply:
Yes, basically this is what we get from the FlagEncoder (plus satisfying the getFlagEncoder method that we'll hopefully remove at a later point). 

But as further investigation showed: we still need and want at least the averageSpeedEnc field because e.g. if we do not tweak the `average_speed` in the FlexModel then we fall back to this value instead of calculating this value from other EncodedValues from scratch for every edge. (this is the "cache effect" I mentioned in some previous discussion)